crates/admin/src/cluster_controller/logs_controller.rs

```diff
@@ -304,49 +305,196 @@ fn build_new_replicated_loglet_configuration(
     previous_configuration: Option<&ReplicatedLogletParams>,
 ) -> Option<ReplicatedLogletParams> {
     let mut rng = thread_rng();
-    // todo make min nodeset size configurable, respect roles, respect StorageState, etc.
-    let nodes_config = metadata().nodes_config_ref();
-
-    let log_servers: Vec<_> = observed_cluster_state
-        .alive_nodes
-        .values()
-        .filter(|node| {
-            nodes_config
-                .find_node_by_id(**node)
-                .ok()
-                .is_some_and(|config| config.has_role(Role::LogServer))
-        })
-        .collect();
+    // todo: make the default nodeset size configurable
+    let replication = previous_configuration
+        .map(|config| config.replication.clone())
+        .unwrap_or_else(|| ReplicationProperty::new(NonZeroU8::new(2).expect("to be valid")));
+
+    let placement = select_log_servers(
```

Comment:
I was imagining creating a single instance of `LogletPlacementSelector` upstream of here, and reusing it for multiple decisions if possible. It's pretty cheap to create though, as it only takes references to nodes config and cluster state.

I don't have any test coverage for this - I've done some manual chaos testing and it works well so far. My plan to improve this with more integrated e2e tests using the `TestCluster` helper as a follow-up.